unstructured_ingest/v2/processes/connectors/duckdb.py

```diff
@@ -0,0 +1,238 @@
+import json
+import uuid
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Callable, Literal, Optional
+
+import pandas as pd
+from pydantic import Field, Secret
+
+from unstructured_ingest.error import DestinationConnectionError
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    FileData,
+    Uploader,
+    UploaderConfig,
+    UploadStager,
+    UploadStagerConfig,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import DestinationRegistryEntry
+
+if TYPE_CHECKING:
+    from duckdb import DuckDBPyConnection as DuckDBConnection
+
+CONNECTOR_TYPE = "duckdb"
+DUCKDB_DB = "duckdb"
+MOTHERDUCK_DB = "motherduck"
+
+
+class MotherDuckAccessConfig(AccessConfig):
```

Comment:
Can we support these as 2 distinct connectors that are based on common base classes? Take a look at this PR I recently opened to update the sql connector that did something similar to support distinct connectors: https://github.com/Unstructured-IO/unstructured-ingest/pull/181